File: easy/2011/2011.ts
import assertEquals from "../../utils/assertEquals";

function finalValueAfterOperations(operations: string[]): number {
    let x = 0;
    for (let i = 0; i < operations.length; i++) {
        switch (operations[i]) {
            case "--X":
                x--;
                break;
            case "X--":
                x--;
                break;
            case "++X":
                x++;
                break;
            case "X++":
                x++;
                break;
        }
    }

    return x;
}

assertEquals(1, finalValueAfterOperations(["X++", "X++", "X--", "++X", "--X"]));

File: easy/2011/2011.py
class Solution(object):
    def finalValueAfterOperations(self, operations):
        """
        :type operations: List[str]
        :rtype: int
        """
        
        x = 0
        for operation in operations:
            if operation == "--X":
                x -= 1
            elif operation == "X--":
                x -= 1
            elif operation == "++X":
                x += 1
            elif operation == "X++":
                x += 1
        return x    

print(Solution().finalValueAfterOperations(["X++", "X++", "X--", "++X", "--X"]) == 1)